main/com/collections/Methods.java

```diff
@@ -137,4 +137,91 @@ public boolean canBePalindrome(String s) {
 
         return arraysSum;
     }
+
+    private void reverseArray(int[] array) {
+        for (int i = 0; i < array.length / 2; i++) {
+            int temp = array[i];
+            array[i] = array[array.length - i - 1];
+            array[array.length - i - 1] = temp;
+        }
+    }
+
+    public int[] sumTwoIntArrays1(int[] array1, int[] array2) {
+
+        // reverse Arrays with private method
+        reverseArray(array1);
+        reverseArray(array2);
+
+        // crate array to store sum
+        ArrayList<Integer> sumOfArrays = new ArrayList<>();
+
+
+        int nextDigit = 0;
+
+        //find which array is bigger and which smaller
+        int lessDigitsAmount = Math.min(array1.length, array2.length);
+        int moreDigitsAmount = Math.max(array1.length, array2.length);
+
+        //first adding part (the same index value in arrays)
+        for (int i = 0; i < lessDigitsAmount; i++) {
+            if ((array1[i] + array2[i] + nextDigit) > 9) {
+                sumOfArrays.add((array1[i] + array2[i] + nextDigit) % 10);
+                nextDigit = 1;
+            } else {
+                sumOfArrays.add(array1[i] + array2[i] + nextDigit);
+                nextDigit = 0;
+
+            }
+
+        }
+
+        //adding rest 1 case (array1.length > array2.length)
+        if (array1.length > array2.length) {
+            for (int i = lessDigitsAmount; i < moreDigitsAmount; i++) {
```

Comment:
You can consider moving this **for** to separated method into which you can pass proper array and other arguments. Such method can be used twice here and below.